core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java

```diff
@@ -84,6 +85,11 @@ public BuildResult run() throws Exception {
                     .addInitial(LaunchModeBuildItem.class)
                     .addInitial(AdditionalApplicationArchiveBuildItem.class)
                     .addInitial(ExtensionClassLoaderBuildItem.class);
+
+            if (!applicationInfo.isUnset()) {
+                chainBuilder.addInitial(ApplicationInfoBuildItem.class);
```

Comment:
This could just as well be produced from a build step which consumes the `ApplicationConfig` object.